examples/Extensions/AddSitelinksUsingFeeds.cs

```diff
@@ -0,0 +1,366 @@
+        /// </summary>
+        /// <param name="args">The command line arguments.</param>
+        public static void Main(string[] args)
+        {
+            AddSitelinksUsingFeeds codeExample = new AddSitelinksUsingFeeds();
+            Console.WriteLine(codeExample.Description);
+
+            // The customer ID for which the call is made.
+            int customerId = int.Parse("INSERT_CUSTOMER_ID_HERE");
+
+            // ID of the campaign to which sitelinks are added.
+            long campaignId = long.Parse("INSERT_CAMPAIGN_ID_HERE");
+
+            // ID of the ad group to which sitelinks are added.
+            // Set to null if you do not wish to limit targeting to a specific ad group.
+            long? adGroupId = long.Parse("INSERT_AD_GROUP_ID_HERE");
+
+            codeExample.Run(new GoogleAdsClient(), customerId, campaignId, adGroupId);
+        }
+
+        /// <summary>
+        /// Returns a description about the code example.
+        /// </summary>
+        public override string Description
+        {
+            get
+            {
+                return "This code example adds sitelinks to a campaign using feed services. Run " +
+                       "AddCampaigns.cs to create a campaign.";
+            }
+        }
+
+        /// <summary>
+        /// Runs the code example.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The customer ID for which the call is made.</param>
+        /// <param name="campaignId">ID of the campaign to which sitelinks are added.</param>
+        /// <param name="adGroupId">Optional ID of the ad group to which sitelinks are added.</param>
+        public void Run(GoogleAdsClient client, long customerId, long campaignId, long? adGroupId)
+        {
+            try
+            {
+                // Create a feed, which acts as a table to store data.  
+                // feed_data = create_feed(client, customer_id)
+                Hashtable feedData = CreateFeed(client, customerId);
+
+                // Create feed items, which fill out the feed table with data.
+                CreateFeedItems(client, customerId, feedData);
+
+                // Create a feed mapping, which tells Google Ads how to interpret this data to
+                // display additional sitelink information on ads.
+                CreateFeedMapping(client, customerId, feedData);
+                
+                // Create a campaign feed, which tells Google Ads which campaigns to use the
+                // provided data with.
+                
+                CreateCampaignFeed(client, customerId, campaignId, feedData);
+                // If an ad group is specified, limit targeting only to the given ad group.
+                if (adGroupId.HasValue)
+                {
+                    CreateAdGroupTargeting(client, customerId, feedData, adGroupId.Value);
```

Comment:
Is ".Value" necessary here?

Reply:
Yes; the value for adGroupId is optional for this example.  The long? type in the method prototype allows it to be null, but introduces the extra step of extracting the value.  Another choice here would be to cast to long, but I think .Value is a little more consistent with the code style.